pkg/uhttp/wrapper.go

```diff
@@ -37,6 +42,52 @@ func WithJSONResponse(response interface{}) DoOption {
 	}
 }
 
+type ErrorResponse interface {
+	Message() string
+}
+
+func WithErrorResponse(resource ErrorResponse) DoOption {
```

Comment:
How this works in parallel with https://github.com/ConductorOne/baton-sdk/pull/110/files#diff-d010464cc4377ff4ba5b4b1c2eb245e880f72188f73564993b55c6546bc56de9R104-R106 ?

Reply:
It should return an error with details before that part of code. I was thinking of the code you sent as a default smallest way for error handling and use this custom option in case you would want to handle error more.